### PR TITLE
apiextensions: structural schema unit test coverage

### PR DESCRIPTION
**Pull request by sttts**
/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go [hunk 1]
```diff
@@ -784,21 +784,23 @@ func (v *specStandardValidatorV3) validate(schema *apiextensions.JSONSchemaProps
...
+					if err := schemaobjectmeta.Coerce(fldPath, clone, s, s.XEmbeddedResource, false); err != nil {
+						allErrs = append(allErrs, err)
+					}
+
+					if !reflect.DeepEqual(clone, interface{}(*schema.Default)) {
+						allErrs = append(allErrs, field.Invalid(fldPath.Child("default"), schema.Default, "must not have unknown fields"))
+					} else if s.XEmbeddedResource {
+						// validate an embedded resource
+						schemaobjectmeta.Validate(fldPath, interface{}(*schema.Default), nil, true)
+					}
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go]: @liggitt take a careful look at this ^

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go [hunk 2]
```diff
@@ -397,7 +416,7 @@ func TestPrune(t *testing.T) {
     }
   }
 }
-`, dontPruneMetaAtRoot: true, schema: &structuralschema.Structural{
+`, isResourceRoot: true, schema: &structuralschema.Structural{
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go]: actually neither the old, nor the new name was good, the old one slightly more correct. What this say is that the prune algorithm will skip the TypeMeta and ObjectMeta fields and leave the pruning to our coercion algorithm.

Comment (liggitt) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go]: should it be `forcePruningAtRoot`?

Comment (liggitt) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go]: what it actually controls is whether the root of the schema is treated as a resource, regardless of the `XEmbeddedResource` setting

**Comment by sttts** [comment]
@roycaihw can you take a look at this?

**Pull request merged**